config/muxcable.py

```diff
@@ -0,0 +1,158 @@
+                    # status is already active, so right back ok
+                    fvs = swsscommon.FieldValuePairs([('status', 'Ok')])
+                    y_cable_response_tbl[asic_index].set(logical_port_name, fvs)
+                    port_status_dict = {'status': 'Ok'}
+                elif status == "active" and status_value == "standby":
+                    # Change of status recived, right back inprogress
+                    fvs = swsscommon.FieldValuePairs([('status', 'inprogress')])
+                    y_cable_response_tbl[asic_index].set(logical_port_name, fvs)
+                    port_status_dict = {'status': 'inprogress'}
+                else:
+                    # Everything else to be treated as failure
+                    fvs = swsscommon.FieldValuePairs([('status', 'Failed')])
+                    y_cable_response_tbl[asic_index].set(logical_port_name, fvs)
+                    port_status_dict = {'status': 'Failed'}
+
+                click.echo("muxcable Ports status : \n {}".format(json.dumps(port_status_dict, indent=4)))
+            else:
+                click.echo("this is not a valid port present on mux_cable".format(port))
+        else:
+            click.echo("there is not a valid asic table for this asic_index".format(asic_index))
+
+    else:
+
+        for namespace in namespaces:
+            asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+            for logical_port in port_table_keys[asic_id]:
+                port_status_dict = {}
+                (status, fvs) = y_cable_asic_table.get(str(logical_port))
+                if status is not True:
+                    click.echo("could not retrieve port values for port".format(logical_port))
+                fvp = dict(fvs)
+                status_value = fvp.get("status", None)
+                if status == "active" and status_value == "active":
+                    # status is already active, so right back ok
+                    fvs = swsscommon.FieldValuePairs([('status', 'Ok')])
+                    y_cable_response_tbl[asic_index].set(logical_port, fvs)
+                    port_status_dict[logical_port] = {'status': 'Ok'}
+                elif status == "active" and status_value == "standby":
+                    # Change of status recived, right back inprogress
+                    fvs = swsscommon.FieldValuePairs([('status', 'inprogress')])
+                    y_cable_response_tbl[asic_index].set(logical_port, fvs)
+                    port_status_dict[logical_port] = {'status': 'inprogress'}
+                else:
+                    # Everything else to be treated as failure
+                    fvs = swsscommon.FieldValuePairs([('status', 'Failed')])
+                    y_cable_response_tbl[asic_index].set(logical_port, fvs)
+                    port_status_dict[logical_port] = {'status': 'Failed'}
+
+                click.echo("muxcable Ports status : \n {}".format(json.dumps(port_status_dict, indent=4)))
```

Comment:
I think we should consider allowing a `--json` flag to output results in JSON format, otherwise output in a human-readable tabular format using `tabulate`.

Reply:
fixed